packages/@react-aria/collections/src/Document.ts

```diff
@@ -256,15 +257,17 @@ export class BaseNode<T> {
  */
 export class ElementNode<T> extends BaseNode<T> {
   nodeType = 8; // COMMENT_NODE (we'd use ELEMENT_NODE but React DevTools will fail to get its dimensions)
-  node: CollectionNode<T>;
+  // TODO: running with assumption that setProps will be called before any other calls to node are made so theoretically
```

Comment:
Don't love this assumption, why did you need to move it out of the constructor? Can we handle most of it but then update specific things later?

Reply:
I actually forget the specifics when I discussed this with @devongovett haha. If we were to use the default CollectionNode here, then the filter function wouldn't be correct during the first collection pass through I think.

Reply:
just to update, chatted with @devongovett and the main reason to drop it out of the constructor was because it was always a kind of intermediate state that would always be updated once the proper types/id/etc came from createLeaf/Branch. Attempting to access a node before setProps is called on it is already an invalid case anyways and I've added a error to catch if that happens like you suggested